ghost/members-api/lib/services/MemberBREADService.js

```diff
@@ -325,6 +325,10 @@ module.exports = class MemberBREADService {
         let model;
 
         try {
+            const suppressionData = await this.emailSuppressionList.getSuppressionData(data.email);
```

Comment:
🤔 Hm, this will get called whenever a member is updated, regardless of whether the their email has changed or not

I was thinking we could save a trip to the db by only doing this if we know the member's email has changed, but we don't have the original member model at this point. We do have [access to it in the repository](https://github.com/TryGhost/Ghost/blob/main/ghost/members-api/lib/repositories/MemberRepository.js#L630), but it would mean we would have to bring `emailSuppressionList` into the repository, which might be tricky / undesirable.

An alternative approach could be to add a new dispatch-able event (`MemberEmailChangeEvent`?) similar to [this](https://github.com/TryGhost/Ghost/blob/main/ghost/members-api/lib/repositories/MemberRepository.js#L623-L627) and move this logic to listen on this event and update accordingly. I suppose the problem with this is the member object returned in the API call won't have the updated `email_disabled` property (as this will be updated in the event handler) so might not be a good fit either?

What do you think? Is it worth the effort to try and move this elsewhere?

Reply:
Yes, great point — I was originally going with a check of whether the email has changed, but this makes things more complicated as you pointed out.

In general, I think it's a good thing to refresh `email_disabled` whenever a member is updated: `email_disabled` and suppressions should be in sync at all times normally, but if they don't, that gives an easy way out for the support team to fix it for a given member.

Reply:
👍 Cool we can leave is for now then